code/modules/projectiles/guns/chaos_bolt.dm

```diff
@@ -0,0 +1,340 @@
+#define CHAOS_STAFF_DAMAGE 30 //Damaging effects will use this number, multiplied or divided depending on effect
+#define CHAOS_STAFF_LETHAL_CHANCE 5 //These should add up to 100
+#define CHAOS_STAFF_NEGATIVE_CHANCE 45
+#define CHAOS_STAFF_MISC_CHANCE 30
+#define CHAOS_STAFF_GIFT_CHANCE 15
+#define CHAOS_STAFF_GREAT_GIFT_CHANCE 5
+
+/obj/item/projectile/magic/chaos
+	name = "chaos bolt"
+	icon_state = "ice_1"
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/chaos
+	var/obj/item/item_to_summon
+	var/explosion_amount = 0 //If left at 0, item goes in backpack or floor, if set, throw that many items around the target
+	var/chaos_effect
+
+/obj/item/projectile/magic/chaos/on_hit(atom/target, blocked = 0)
+	. = ..()
+
+	if(iswallturf(target) || isobj(target))
+		target.color = pick(GLOB.random_color_list)
+
+	if(target && isliving(target))
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			L.visible_message("<span class='warning'>[target] glows faintly, but nothing else happens.</span>")
+			return
+		chaos_chaos(target)
+
+/obj/item/projectile/magic/chaos/Initialize(mapload)
+	. = ..()
+	icon_state = pick("bluespace", "pulse1", "magicm", "declone", "fireball", "blood_bolt", "arcane_barrage", "laser", "u_laser")
+
+/obj/item/projectile/magic/chaos/proc/chaos_chaos(mob/living/target, blocked = 0)
+	var/category = pick(prob(CHAOS_STAFF_LETHAL_CHANCE);"lethal", prob(CHAOS_STAFF_NEGATIVE_CHANCE);"negative", prob(CHAOS_STAFF_MISC_CHANCE);"misc",\
+		prob(CHAOS_STAFF_GIFT_CHANCE);"gift", prob(CHAOS_STAFF_GREAT_GIFT_CHANCE);"great gift")
+	switch(category)
+		if("lethal") //Target is either dead on the spot or might as well be
+			apply_lethal_effect(target)
+		if("negative") //Target is damaged, crippled or otherwise negatively affected. Effect can be lethal in some circumstances.
+			apply_negative_effect(target)
+		if("misc") //Miscellaneous effect, can be positive, negative, or just humorous
+			apply_misc_effect(target)
+		if("gift") //Grants a gift or positive effect to the target. Usually a gag or mildly useful item... if you weren't being killed by a wizard
+			apply_gift_effect(target)
+		if("great gift") //Grants a gift or positive effect to the target. Usually a weapon or useful item.
+			apply_great_gift_effect(target)
+	if(item_to_summon) //TODO check if mob's alive, no effect on dead mobs
```

Comment:
have yet to check it?

Reply:
has been handled, forgot to remove comment, removing